pcf/test/particle/aws/cloudfront/test_cloudfront.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import moto
+
+from pcf.particle.aws.cloudfront.cloudfront import CloudFront
+from pcf.core import State
+
+
+class TestCloudFront:
+    particle_definition = {
+        "pcf_name": "pcf_cloudfront",
+        "flavor": "cloudfront",
+        "aws_resource": {
+            "Tags": [
+                {
+                    "Key": "Name",
+                    "Value": "cloud-front1"
+                }
+            ],
+            "Comment": "pcf-cloud-front", # used as the uid bc getting a distribution and its tags are separate api calls
+            "CallerReference": "sdfa6df5a4sdf5asd7f9asd6fa5sd6f7a8sd9fa8sdf",
+            "Origins": {
+                "Quantity": 1,
+                "Items": [
+                    {
+                        "Id": "wahoo",
+                        "DomainName": "hoo.com",
+                    },
+                ]
+            },
+            "DefaultCacheBehavior": {
+                "TargetOriginId": "string",
+                "ForwardedValues": {
+                    "QueryString": False,
+                    "Cookies": {
+                        "Forward": "none",
+                    },
+                },
+                "TrustedSigners": {
+                    "Enabled": True,
+                    "Quantity": 12,
+                },
+                "ViewerProtocolPolicy": "allow-all",
+                "MinTTL": 50,
+            },
+            "Enabled": True,
+        }
+    }
+    """No cloudfront implementaiton in moto"""
```

Comment:
while there may not be a cloudfront implementation in moto, asserting true here seems to be the wrong move here as this test is essentially useless.  The Custodian project uses placebo (https://github.com/garnaat/placebo) to record and replay http responses from AWS. Another option to do test recording is vcr.py (https://github.com/kevin1024/vcrpy) . Either of these options allows you to test new features from aws without relying on an implementation from another library first

Reply:
awesome yeah those look like good options. we tried just monkey patching and it was a huge pain. will check those libraries out!